### PR TITLE
new token field and double check validation

### PR DESCRIPTION
**Pull request by rmarinleal**
* Add token field in pagantis orders table
* Prevent multitab order creation

### CODE REVIEW

File: controllers/front/notify.php [hunk 1]
```diff
@@ -273,8 +279,8 @@ public function prepareVariables()
     public function getMerchantOrderId()
     {
         try {
-            $table = _DB_PREFIX_ .self::ORDERS_TABLE;
-            $sql = 'select ps_order_id from ' . $table .' where id = ' .(int)$this->merchantCartId;
+            $sql = 'select ps_order_id from ' . _DB_PREFIX_ .self::ORDERS_TABLE .' where id = '
+                .(int)$this->merchantCartId . ' and token = \'' . $this->token . '\'';
```
Comment (pgarcess) [inline controllers/front/notify.php:283]: scaping token parameter? why not id? Remove both?

File: controllers/front/notify.php [hunk 2]
```diff
@@ -318,9 +324,9 @@ public function getMerchantOrder()
     private function getPagantisOrderId()
     {
         try {
-            $this->pagantisOrderId= Db::getInstance()->getValue(
-                'select order_id from '._DB_PREFIX_.self::ORDERS_TABLE.' where id = '.(int)$this->merchantCartId
-            );
+            $sql = 'select order_id from ' . _DB_PREFIX_.self::ORDERS_TABLE .' where id = '
+                .(int)$this->merchantCartId . ' and token = \'' . $this->token . '\'';
```
Comment (pgarcess) [inline controllers/front/notify.php:328]: scaping token parameter? why not id? Remove both?

File: controllers/front/payment.php [hunk 3]
```diff
@@ -434,4 +435,4 @@ private function getUserLanguage($shippingAddress = null, $billingAddress = null
         }
         return 'ES';
     }
-}
+}
```
Comment (pgarcess) [inline controllers/front/payment.php]: Add a new line

File: controllers/front/notify.php (hunk 1 above)
Comment (rmarinleal) [reply controllers/front/notify.php:283]: I'm preserving simple quote for string sql parameter not scaping

File: controllers/front/notify.php (hunk 2 above)
Comment (rmarinleal) [reply controllers/front/notify.php:328]: I'm preserving simple quote for string sql parameter not scaping

**Review by pgarcess — APPROVED** [review]

File: controllers/front/payment.php (hunk 3 above)
Comment (rmarinleal) [reply controllers/front/payment.php]: changed ;)

**Pull request merged**